src/Utils/ProductGalleryUtils.php

```diff
@@ -33,17 +33,21 @@ public static function get_product_gallery_images( $post_id, $size = 'full', $at
 				$size = $crop_images ? self::CROP_IMAGE_SIZE_NAME : $size;
 
 				foreach ( $all_product_gallery_image_ids as $product_gallery_image_id ) {
-					if ( $crop_images ) {
-						self::maybe_generate_intermediate_image( $product_gallery_image_id, self::CROP_IMAGE_SIZE_NAME );
+					if ( '0' !== $product_gallery_image_id ) {
+						if ( $crop_images ) {
+							self::maybe_generate_intermediate_image( $product_gallery_image_id, self::CROP_IMAGE_SIZE_NAME );
+						}
+
+						$product_image_html = wp_get_attachment_image(
+							$product_gallery_image_id,
+							$size,
+							false,
+							$attributes
+						);
+					} else {
+						$product_image_html = self::get_product_image_placeholder_html( $attributes );
```

Comment:
Instead of creating our own function here perhaps use what's already available from WC core. `wc_placeholder_img( 'full', $attributes );`